ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Stream.scala

```diff
@@ -411,30 +411,9 @@ private[h2] class H2Stream[F[_]: Concurrent](
   def getRequest: F[org.http4s.Request[fs2.Pure]] = state.get.flatMap(_.request.get.rethrow)
   def getResponse: F[org.http4s.Response[fs2.Pure]] = state.get.flatMap(_.response.get.rethrow)
 
-  def readBody: Stream[F, Byte] = {
-    def pullBuffer(buffer: Queue[F, Either[Throwable, ByteVector]]): Pull[F, Byte, Unit] =
-      Pull.eval(buffer.tryTake).flatMap {
-        case Some(Right(s)) => Pull.output(Chunk.byteVector(s)) >> pullBuffer(buffer)
-        case Some(Left(e)) => Pull.raiseError(e)
-        case None => Pull.done
-      }
```

Comment:
... but before the data was offered to the `buffer`. So the `tryTake` would come up empty and the read stream would be terminated prematurely.

Reply:
I wonder if this can carry consequences for some other parts of Ember 🤔